cps/helper.py

```diff
@@ -428,32 +428,46 @@ def delete_book(book, calibrepath, book_format):
         return delete_book_file(book, calibrepath, book_format)
 
 
+def get_cover_on_failure(use_generic_cover):
+    if use_generic_cover:
+        return send_from_directory(_STATIC_DIR, "generic_cover.jpg")
+    else:
+        return None
+
 def get_book_cover(book_id):
     book = db.session.query(db.Books).filter(db.Books.id == book_id).first()
-    if book.has_cover:
+    return get_book_cover_internal(book, False)
```

Comment:
With use generic cover on failure I get this error:
  File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 2097, in make_response
    "The view function did not return a valid response. The"
TypeError: The view function did not return a valid response. The function either returned None or ended without a return statement.
Cause are calls to non existing covers, in this case "get_cover_on_failure" returns None. The parameter should be true. Could you please doublecheck this.

Reply:
Ah yeah that's correct, in fact that was the whole point of that refactor :).
Fixed.